domain/infosync/info.go

```diff
@@ -699,12 +693,7 @@ func (is *InfoSyncer) getPrometheusAddr() (string, error) {
 	}
 
 	// Get prometheus address from pdApi.
-	var url, res string
-	if strings.HasPrefix(pdAddrs[0], "http://") {
-		url = fmt.Sprintf("%s%s", pdAddrs[0], pdapi.Config)
-	} else {
-		url = fmt.Sprintf("http://%s%s", pdAddrs[0], pdapi.Config)
-	}
+	url := util2.ComposeURL(pdAddrs[0], pdapi.Config)
 	resp, err := http.Get(url) // #nosec G107
```

Comment:
When `https` scheme is used, the `http.Get(url)` reports an error. 

Please try to use https://github.com/pingcap/tidb/issues/26568#issuecomment-887426960 instead of `http.Get()`.